plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/ValueAdjuster.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.util;
+
+import io.trino.spi.block.Block;
+import io.trino.spi.type.Type;
+
+import java.util.function.Function;
+
+import static java.util.Objects.requireNonNull;
+
+public abstract class ValueAdjuster<T extends Type>
+        implements Function<Block, Block>
+{
+    protected final T forType;
+
+    protected ValueAdjuster(T forType)
+    {
+        this.forType = requireNonNull(forType);
+    }
+
+    @Override
+    public abstract Block apply(Block block);
```

Comment:
`applyBlock` mostly looks same - we iterate over its position `checkForNull` and apply the translation for non null values - Can we create an abstraction only for it ?